ci-operator/step-registry/openshift-logging/deploy-operators/openshift-logging-deploy-operators-commands.sh

```diff
@@ -0,0 +1,526 @@
+    # if [[ ${caps} =~ "marketplace" ]]; then
+    #     echo "marketplace installed, skip..."
+    #     return 0
+    # fi
+    ret=0
+    run_command "oc get ns openshift-marketplace" || ret=$?
+    if [[ $ret -eq 0 ]]; then
+        echo "openshift-marketplace project AlreadyExists, skip creating."
+        return 0
+    fi
+
+    cat <<EOF | oc create -f -
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+    security.openshift.io/scc.podSecurityLabelSync: "false"
+    pod-security.kubernetes.io/enforce: baseline
+    pod-security.kubernetes.io/audit: baseline
+    pod-security.kubernetes.io/warn: baseline
+  name: openshift-marketplace
+EOF
+}
+
+# from OCP 4.15, the OLM is optional, details: https://issues.redhat.com/browse/OCPVE-634
+# since OCP4.18, OLMv1 is a new capability: OperatorLifecycleManagerV1
+function check_olm_capability() {
+    # check if OLMv0 capability is added
+    knownCaps=$(oc get clusterversion version -o=jsonpath="{.status.capabilities.knownCapabilities}")
+    if [[ ${knownCaps} =~ "OperatorLifecycleManager\"," ]]; then
+        echo "knownCapabilities contains OperatorLifecycleManagerv0"
+        # check if OLMv0 capability enabled
+        enabledCaps=$(oc get clusterversion version -o=jsonpath="{.status.capabilities.enabledCapabilities}")
+        if [[ ! ${enabledCaps} =~ "OperatorLifecycleManager\"," ]]; then
+            echo "OperatorLifecycleManagerv0 capability is not enabled, skipping operator installation."
+            return 1 # Return a non-zero exit code
+        fi
+    fi
+    return 0 # Return 0 for success
+}
+
+set_proxy
+# Check for required commands
+command -v jq >/dev/null 2>&1 || { echo >&2 "Error: jq is not installed. Aborting."; exit 1; }
+command -v yq-go >/dev/null 2>&1 || { echo >&2 "Error: yq-go is not installed. Aborting."; exit 1; }
+
+run_command "echo $PATH"
+run_command "ls -l /cli"
+run_command "which oc && oc version -o yaml"
+run_command "oc whoami"
+update_global_auth
+sleep 5
+create_icsp_connected
+
+check_olm_capability
+if [[ $? -ne 0 ]]; then
+    echo "Skipping operator installation due to OLM capability."
+    exit 0
+fi
+
+if [[ ! -z ${index_image} ]]; then
+    check_marketplace
+    create_catalog_sources
+    subscribe_operator "cluster-logging" "openshift-logging"
+    subscribe_operator "loki-operator" "openshift-operators-redhat"
+else
+    echo "Install operator via operator-sdk"
+    # install operator-sdk
+    export HOME=/tmp/home
+    mkdir -p "${HOME}"
+    export PATH=$PATH:$HOME
+
+    cd $HOME
+
+    ARCH=$(case $(uname -m) in x86_64) echo -n amd64 ;; aarch64) echo -n arm64 ;; *) echo -n "$(uname -m)" ;; esac)
+    export ARCH
+    OS=$(uname | awk '{print tolower($0)}')
+    export OS
+
+    export OPERATOR_SDK_DL_URL=https://github.com/operator-framework/operator-sdk/releases/download/v1.41.1
+    curl -L ${OPERATOR_SDK_DL_URL}/operator-sdk_${OS}_${ARCH} -o operator-sdk
```

Comment:
can you echo "curl -L ${OPERATOR_SDK_DL_URL}/operator-sdk_${OS}_${ARCH} -o operator-sdk" ? so we can try this script when it fail